src/cascadia/TerminalApp/App.base.h

```diff
@@ -9,30 +9,23 @@ namespace winrt::TerminalApp::implementation
 
         IXamlType GetXamlType(::winrt::Windows::UI::Xaml::Interop::TypeName const& type)
         {
-            return AppProvider()->GetXamlType(type);
+            return AppProvider().GetXamlType(type);
         }
 
         IXamlType GetXamlType(::winrt::hstring const& fullName)
         {
-            return AppProvider()->GetXamlType(fullName);
+            return AppProvider().GetXamlType(fullName);
         }
 
         ::winrt::com_array<::winrt::Windows::UI::Xaml::Markup::XmlnsDefinition> GetXmlnsDefinitions()
         {
-            return AppProvider()->GetXmlnsDefinitions();
+            return AppProvider().GetXmlnsDefinitions();
         }
 
     private:
         bool _contentLoaded{ false };
-        std::shared_ptr<XamlMetaDataProvider> _appProvider;
-        std::shared_ptr<XamlMetaDataProvider> AppProvider()
-        {
-            if (!_appProvider)
-            {
-                _appProvider = std::make_shared<XamlMetaDataProvider>();
-            }
-            return _appProvider;
-        }
+        winrt::TerminalApp::XamlMetaDataProvider _appProvider = winrt::TerminalApp::XamlMetaDataProvider();

```

Comment:
nit: This can be
```cpp
winrt::TerminalApp::XamlMetaDataProvider _appProvider;
```

The default constructor creates an instance of the given type automatically. This is called [uniform construction](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/consume-apis#uniform-construction).

Reply:
True, updated that now.